data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,236 @@
+#!/bin/bash
+
+#PREPARATION
+#Usage and help info for the script
+function help_usage() {
+
+    echo "script usage: $(basename $0) [-s Start Monitor for guest console] [-e End of Monitor for guest console ] [-h help]"
+}
+
+monitor_log_file="/tmp/guest_console_monitor.log"
+#Quit if there are less than one argument
+if [ $# -eq 0 ];then
+    help_usage | tee -a ${monitor_log_file}
+    exit 1
+fi
+
+##CONFIG PART
+START_MONITOR="0"
+END_MONITOR="0"
+MONITOR_LOCKED_FILE="/tmp/GUEST_CONSOLE_COLLECTOR_SEMAPHORE"
+ALL_CMD="guest_console_monitor.sh -s|expect -c|virsh console"
+LEGACY_CMD="expect -c|virsh console"
+LOG_DIR="/tmp/virt_logs_residence"
+###CHECK EXISTED GUEST
+vm_guestnames_types="sles"
+get_vm_guestnames_inactive=`virsh list --inactive | grep "${vm_guestnames_types}" | awk '{print $2}'`
+vm_guestnames_inactive_array=$(echo -e ${get_vm_guestnames_inactive})
+get_vm_guestnames=`virsh list  --all | grep "${vm_guestnames_types}" | awk '{print $2}'`
+vm_guestnames_array=$(echo -e ${get_vm_guestnames})
+vmguest=""
+vmguest_failed="0"
+
+#Parse input arguments. -s or -e must have values
+while getopts "seh" OPTION; do
+  case "${OPTION}" in
+    s)
+      START_MONITOR="1"
+      ;;
+    e)
+      END_MONITOR="1"
+      ;;
+    h)
+      help_usage | tee -a ${monitor_log_file}
+      exit 1
+      ;;
+    *)
+      help_usage | tee -a ${monitor_log_file}
+      exit 1
+      ;;
+  esac
+done
+
+#FUNCTION PART
+function get_console() {
+
+local vmguest=$1
+
+expect -c "
+set timeout 60
+
+##hide echo
+log_user 0
+spawn -noecho virsh console ${vmguest}
+
+#wait connection
+sleep 3
+send \"\r\n\r\n\r\n\"
+
+#condition expect
+expect {
+        \"*login:\" {
+                send \"root\r\"
+                exp_continue
+        }
+        -nocase \"password:\" {
+                send \"novell\r\"
+                exp_continue
+        }
+        \"*:~ #\" {
+                send -- \"ip route get 1\r\"
+        }
+        timeout {
+                send -- \"exit\r\"
+                exp_continue        
+        }                 
+}
+
+## -1 means never timeout
+set timeout -1
+
+expect -re {dev.*\s([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})[^0-9]}
+
+if {![info exists expect_out(1,string)]} {
+        puts \"Match did not happen :(\"
+}
+set output \$expect_out(buffer)
+expect eof
+
+puts \"\$output\"
+"
+
+}
+
+function open_console() {
+
+    #open guest console
+    for vmguest in ${vm_guestnames_state_running_array[@]};do
+        get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console.log 2>&1 &
+    done 
+}
+
+function close_console() {
+
+    key_cmd=$@
+    #close guest console
+    count=0
+    until [ "$count" -eq 1 ]
```

Comment:
This until loop is not needed if you only try once kill -9

Reply:
sure, got it.

Reply:
rename this stop_sign as close_console_count, also, add the related comments for this close_console_count
DONE